### PR TITLE
Feature/sanity studio

### LINKED ISSUE #139: Connect Shopify to Sanity using the Sanity Connect for Shopify
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/137

### Description

na

### LINKED ISSUE #144: [S2.1]: Install Sanity CMS into the ecommerce package
### Link to Epic

Epic: https://github.com/Open-Study-College/osc/issues/143

### Description

N/A

### Linked Tasks

- [x] #145
- [x] #146 
- [x] #147 
- [x] #148 
- [x] #149 
- [x] #204

### LINKED ISSUE #145: Store API tokens in .env
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/144

### Description

n/A

### LINKED ISSUE #146: Setup Sanity config file
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/144

### Description

- add basePath
- add projectID
- add dataset

### LINKED ISSUE #147: Configure new document options
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/144

### Description

Lock singletons from being crated (home/settings)

### LINKED ISSUE #148: Configure plugins
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/144

### Description

- dashboard
- deskTool
- visionTool
- media

### LINKED ISSUE #149: Configure custom actions
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/144

### Description

- Delete shopify items warning
- Duplicate shopify items warning

### LINKED ISSUE #204: Configure document types
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/144

### Description

- Pages
- blog posts
- courses (Products)
- Collections
- Navigation

### PR DESCRIPTION
**Pull request by stevenbister**
- Resolves #139 
- Resolves #144 
- Resolves #145 
- Resolves #146 
- Resolves #147 
- Resolves #148 
- Resolves #149 
- Resolves #204 

## 📝 Description

Apologies for the large PR, covers a lot off in this one!

### `osc-studio`
- Adds Sanity Studio to the packages directory within the monorepo
- Adds Sanity variables to my local .env and have updated the .env.sample and README to include them
- Links Sanity with Shopify and sets up the documents required
- Sets up initial custom actions and plugins

### `osc-ecommerce`
- Adds Sanity Client to query the data from our content lake.
- Adds some initial routes to pass data to (I imagine these might change slightly in naming)
- Adds preview component and hook which will run when on a page with `?preview=true` in the url

## ⛳️ Current behavior (updates)

Sans Sanity

## 🚀 New behavior

Adds Sanity to the packages and queries to `osc-ecommerce`

## 📝 Additional Information

You will need the Sanity ID and API keys if you don't already have them!

I'll create a separate PR for building a URL for the Sanity Studio as this one is pretty big already without me testing that out 👀

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,7 +95,8 @@
     "dev": "lerna run dev",
     "dev:osc-ecommerce": "lerna run dev --scope '{osc-ui,osc-ecommerce}'",
     "dev:osc-academic-hub": "lerna run dev --scope '{osc-ui,osc-academic-hub}'",
-    "setup": "lerna link convert && npm i && npx prisma generate"
+    "dev:osc-studio": "cd packages/osc-studio && npm run dev",
+    "setup": "lerna link convert && npm i && npx prisma generate && cd packages/osc-studio && npm i --legacy-peer-deps"
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply package.json:99]: Unfortunately yes (for now) some of the Sanity plugins are dependant on it to run correctly. We'll be able to remove it once we upgrade to V3. 
Specifically It's related to the dashboard, so I guess we could remove it but I think it's a nice to have.

Is it interfering with anything?

File: packages/osc-studio/utils/resolveProductionUrl.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+const localUrl = `http://localhost:2000`;
+// TODO: Update this
+const remoteUrl = `https://your-deployed-website.com`;
```
Comment (hi-im-si) [inline packages/osc-studio/utils/resolveProductionUrl.ts:3]: I am guessing it might be worth moving this to an ENV variable

**Review by hi-im-si — COMMENTED** [review]
LGTM!

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (hi-im-si) [reply package.json:99]: I am guessing this is to not install any older version dependencies?

File: packages/osc-studio/utils/resolveProductionUrl.ts (hunk 2 above)
Comment (stevenbister) [reply packages/osc-studio/utils/resolveProductionUrl.ts:3]: Yeah for sure, I think I'll set something up in a separate PR to build the studio so I'll know what urls I'll need

**Review by hi-im-si — APPROVED** [review]

**Pull request merged**